src/crystal/system/unix/process.cr

```diff
@@ -59,7 +59,13 @@ struct Crystal::System::Process
   end
 
   def self.on_interrupt(&handler : ->) : Nil
-    ::Signal::INT.trap { |_signal| handler.call }
+    sig_handler = Proc(::Signal, Nil).new do |signal|
+      handler.call
+      signal.ignore
```

Comment:
It feels more correct to me to inverse the calls order, since you'd like to ignore the signal regardless of the `handler` returning/raising.

```suggestion
      signal.ignore
      handler.call
```

Reply:
The windows version relies on the return value of the function so if an error is raised then the signal won't be ignored.
So ignoring after the handler call is more accurate